src/pkg/services/m365/api/groups.go

```diff
@@ -50,6 +51,25 @@ func (c Groups) GetAll(
 	return getGroups(ctx, errs, service)
 }
 
+// GetAllIDsAndNames retrieves all groups in the tenant and returns them in an idname.Cacher
+func (c Groups) GetAllIDsAndNames(ctx context.Context, errs *fault.Bus) (idname.Cacher, error) {
```

Comment:
Request: can you please move this addition to a separate PR?  And, in that change, can you please remove this old func in services favor of this new one? https://github.com/alcionai/corso/blob/main/src/pkg/services/m365/groups.go#L160

Reply:
Moving this out to a separate PR might get a bit messy as we need it for this PR, but let me create a follow up to remove the old one.

Reply:
Created https://github.com/alcionai/corso/pull/4967